packages/manager/src/features/CloudPulse/Alerts/CreateAlert/Criteria/DimensionFilterValue/useCleanupInvalidateValues.ts

```diff
@@ -0,0 +1,55 @@
+// useCleanupInvalidValues.ts
+import { useEffect } from 'react';
+
+import type { Item } from '../../../constants';
+
+/**
+ * Cleans up stale form values that are no longer in the options list.
+ */
+export const useCleanupInvalidValues = ({
```

Comment:
```suggestion
export const useCleanupStaleValues = ({
```

this would be much better name, invalid is very ambiguous

Reply:
@nikhagra-akamai , I will rename it. The testcases are covered in the relevant DimensionFilterAutocomplete components for Firewall and ObjectStorage.